Library/Homebrew/dependencies.rb

```diff
@@ -123,3 +123,61 @@ def reject_ignores(dependables, ignores, includes)
     end
   end
 end
+
+# An adapter for casks to provide dependency information in a formula-like interface
+class CaskDependent
```

Comment:
I'd be tempted to say a `cask_dependent` file or `cask/dependent` would be even nicer. A few unit tests then would be nice, too!